src/outputs/outputs.cpp

```diff
@@ -252,20 +261,21 @@ Outputs::Outputs(Mesh *pm, ParameterInput *pin, SimTime *tm) {
         pnew_type = new FormattedTableOutput(op);
       } else if (op.file_type.compare("vtk") == 0) {
         pnew_type = new VTKOutput(op);
-      } else if (op.file_type.compare("rst") == 0) {
-        pnew_type = new RestartOutput(op);
-        num_rst_outputs++;
-      } else if (op.file_type.compare("ath5") == 0 || op.file_type.compare("hdf5") == 0) {
+      } else if (is_hdf5_output) {
+        bool restart = (op.file_type.compare("rst") == 0);
```

Comment:
make restart const. 

Also I think that comparing strings with == is a little bit more readable. Is there a reason you use compare? 

```
op.file_type == "rst"
```

Reply:
I agree. I don't know who wrote the initial version using `compare`.